TileStache/Goodies/VecTiles/transform.py

```diff
@@ -1156,6 +1159,283 @@ def exterior_boundaries(feature_layers, zoom,
+
+    elif geom_type == 'MultiLineString':
+        return linemerge(geom)
+
+    else:
+        return MultiLineString([])
+
+
+def admin_boundaries(feature_layers, zoom, base_layer,
+                     start_zoom=0):
+    """
+    Given a layer with admin polygons and maritime boundaries,
+    attempts to output a set of oriented boundaries with properties
+    from both the left and right polygon, and also cut with the
+    maritime information to provide a `maritime_boundary=yes` value
+    where there's overlap between the maritime lines and the
+    polygon boundaries.
+    """
+
+    layer = None
+
+    # don't start processing until the start zoom
+    if zoom < start_zoom:
+        return layer
+
+    layer = _find_layer(feature_layers, base_layer)
+    if layer is None:
+        return None
+
+    # layer will have polygonal features for the admin
+    # polygons and also linear features for the maritime
+    # boundaries. further, we want to group the admin
+    # polygons by their kind, as this will reduce the
+    # working set.
+    admin_features = defaultdict(list)
+    maritime_features = list()
+    new_features = list()
+
+    for shape, props, fid in layer['features']:
+        dims = _geom_dimensions(shape)
+        kind = props.get('kind')
+        maritime_boundary = props.get('maritime_boundary')
+
+        # note: dims=4 means polygon, as it's a bit set of
+        # 1 << num_dimensions. likewise, dims=2 means a
+        # linestring. the reason to use this rather than
+        # compare the string of types is to catch the
+        # "multi-" types as well.
+        if dims == 4 and kind is not None:
+            admin_features[kind].append((shape, props, fid))
+
+        elif dims == 4 and maritime_boundary == 'yes':
+            maritime_features.append((shape, {'maritime_boundary':'no'}, 0))
+
+    # there are separate polygons for each admin level, and
+    # we only want to intersect like with like because it
+    # makes more sense to have Country-Country and
+    # State-State boundaries (and labels) rather than the
+    # (combinatoric) set of all different levels.
+    for kind, features in admin_features.iteritems():
+        num_features = len(features)
+        envelopes = map(lambda g: g[0].envelope, features)
+
+        for i, feature in enumerate(features):
+            shape, props, fid = feature
+            envelope = envelopes[i]
+
+            # orient to ensure that the shape is to the
+            # left of the boundary.
+            boundary = orient(shape).boundary
+
+            # intersect with *preceding* features to remove
+            # those boundary parts. this ensures that there
+            # are no duplicate parts.
+            for j in range(0, i):
+                cut_shape, cut_props, cut_fid = features[j]
+                cut_envelope = envelopes[j]
+                if envelope.intersects(cut_envelope):
+                    boundary = _intersect_cut(boundary, cut_shape)[1]
```

Comment:
Is it better to just use the difference explicitly here to try and save on some processing?

```python
boundary = boundary.difference(cut_shape)
```